clang/lib/AST/DeclCXX.cpp

```diff
@@ -1526,10 +1526,9 @@ bool CXXRecordDecl::isGenericLambda() const {
 
 #ifndef NDEBUG
 static bool allLookupResultsAreTheSame(const DeclContext::lookup_result &R) {
-  for (auto *D : R)
-    if (!declaresSameEntity(D, R.front()))
-      return false;
-  return true;
+  return llvm::all_of(R, [&](NamedDecl *D) {
+    return D->isInvalidDecl() || declaresSameEntity(D, R.front());
+  });
```

Comment:
Should we move the isInvalidDecl check to declaresSameEntity?
I doubt this is the only place this might happen

Reply:
From what I can tell nowhere is checking what they pass to `declaresSameEntity(...)` but I did not dig too deeply to see if we know they are valid earlier in some way.